project-koku/koku
# Cost 2605 rank summation tune up

Fixes [COST-2605](https://issues.redhat.com/browse/COST-2605)

Changes proposed in this PR:
* Tune-up `_perform_rank_summation` method code and called functions for speed

Testing Instructions:
1. checkout branch
2. Generate a large amount of AWS data (like, **really** big; like >= 10k AWS users big)
3. Hit the aws cost explorer endpoint with something like:
    `http://127.0.0.1:8000/api/cost-management/v1/reports/aws/costs/?filter[limit]=5&group_by[account]=*&start_date=2022-03-01&end_date=2022-05-02`
4. Examine the `koku-server` logs. You should see a log entry with the text `_perform_rank_summation total time` The decimal number you see _should_ be sub-second completion
